cirrus/google_cloud/iam.py

```diff
@@ -213,7 +228,13 @@ def __str__(self):
         Returns:
             str: Representation of the Role for Google's API
         """
-        return "{}{}".format(GooglePolicyRole.ROLE_PREFIX, self.name)
+        # / means the role already has a prefix in the name
+        if "/" in self.name:
```

Comment:
why did you move this check from `__init__ ` to `__str__ `?
you also removed the `.strip()`s in `__init__` so maybe the name can be set incorrectly?

I think this test should also check that `name` is set correctly: https://github.com/uc-cdis/cirrus/blob/master/test/test_google_cloud_iam.py#L9

Reply:
google custom roles require a different prefix with "/", being "projects/project-name/roles/". so the first change was to stop trying to be smart about the role prefix during initialization and just set it to what was given. then, when converting to a valid string, if it's a normal role (e.g. no "/") preprend the necessary prefix

Reply:
that being said, if we want to maintain exact backwards compatibility, we could add the code back in that does the initial name sanitation (removing the prefix), but we'd still need this chunk of code